demonstrations/tutorial_vqt.py

```diff
@@ -0,0 +1,825 @@
+# expectation exactly for each iteration. this is not how the VQT would
+# work in the real world, for large systems where the number of basis
+# states (and thus the size of the probability distribution) scales
+# exponentially, but for small toy-models such as this, the exact form
+# runs faster:
+#
+
+
+def exact_cost(params):
+
+    global iterations
+
+    # Separates the list of parameters
+
+    dist_params = params[0:qubit]
+    params = params[qubit:]
+
+    # Creates the probability distribution
+
+    distribution = prob_dist(dist_params)
+
+    # Generates a list of all computational basis states, of our qubit system
+
+    s = [[int(i) for i in list(bin(k)[2:].zfill(qubit))] for k in range(0, 2 ** qubit)]
+
+    # Passes each basis state through the variational circuit and multiplis the calculated energy EV
+    # with the associated probability from the distribution
+
+    final_cost = 0
+    for i in s:
+        result = qnode(params, qubits, i, 3)
+        for j in range(0, len(i)):
+            result = result * distribution[j][i[j]]
+        final_cost += result
+
+    # Calculates the entropy and the final cost function
+
+    entropy = calculate_entropy(distribution)
+    final_final_cost = beta * final_cost - sum(entropy)
+
+    if iterations % 50 == 0:
+        print("Cost at Step " + str(iterations) + ": " + str(final_final_cost))
+
+    iterations += 1
+
+    return final_final_cost
+
+
+######################################################################
+# Finally, we optimize the cost function:
+#
+
+# Creates the optimizer
+
+iterations = 0
+params = [random.randint(-100, 100) / 100 for i in range(0, (12 * depth) + qubit)]
+
+out = minimize(exact_cost, x0=params, method="COBYLA", options={"maxiter": 1000})
+params = out["x"]
+print(out)
+
+
+######################################################################
+# With our optimal parameters, we now wish to prepare the state to which
+# they correspond, to see "how close" our prepared state is to the target
+# state. This can be done by simply taking the optimal parameters, and
+# passing each possible basis state through the variational circuit. Each
+# corresponding probability is multiplied by the outer product of the
+# resulting state with itself. Once we add these all together, we are left
+# with the density matrix corresponding to the optimal parameters.
+#
+
+
+def prepare_state(params, device):
+
+    # Initializes the density matrix
+
+    final_density_matrix_2 = np.zeros((2 ** qubit, 2 ** qubit))
+
+    # Prepares the optimal parameters, creates the distribution and the bitstrings
+
+    dist_params = params[0:qubit]
+    unitary_params = params[qubit:]
+
+    distribution = prob_dist(dist_params)
+
+    s = [[int(i) for i in list(bin(k)[2:].zfill(qubit))] for k in range(0, 2 ** qubit)]
+
+    # Runs the circuit in the case of the optimal parameters, for each bitstring, and adds the result to the final density matrix
```

Comment:
Adding in a line break here to avoid horizontal scrolling on the website 
```suggestion
    # Runs the circuit in the case of the optimal parameters, for each bitstring,
    # and adds the result to the final density matrix.
```